app/src/androidTest/java/org/mozilla/fenix/perf/HomeFragmentBenchmark.kt

```diff
@@ -0,0 +1,150 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.content.Intent
+import android.graphics.Point
+import android.view.View
+import android.view.ViewGroup
+import androidx.benchmark.junit4.BenchmarkRule
+import androidx.benchmark.junit4.measureRepeated
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import androidx.test.uiautomator.By
+import org.junit.Assert.assertNotNull
+import mozilla.components.browser.toolbar.BrowserToolbar
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.IntentReceiverActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.helpers.TestHelper
+import org.mozilla.fenix.home.HomeFragment
+import org.mozilla.fenix.ui.robots.mDevice
+import org.mozilla.fenix.utils.view.ViewHolder
+
+@RunWith(AndroidJUnit4::class)
+class HomeFragmentBenchmark {
+
+    private val sessionControlViewResourceId = "sessionControlRecyclerView"
+
+    private val homeButtonContentDescription = "Home screen"
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val benchmarkRule = BenchmarkRule()
+
+    @get:Rule
+    val homeActivityRule = HomeActivityTestRule(
+        skipOnboarding = true, skipShouldShowJumpBackIn = true
+    )
+
+    @get: Rule
+    val intentReceiverActivityTestRule = ActivityTestRule(
```

Comment:
thought: `ActivityTestRule` is deprecated in favor of `ActivityScenario`. I don't think it's worth our time to change it now but FYI for the future